### ISSUE TITLE
James's shadow bug at different resolutions

### ISSUE THREAD

**Issue by Nikatomix** [elishacloud/Silent-Hill-2-Enhancements#1089]
Video: https://youtu.be/xwYbTczSwZo

There's a shadowy outline on James's right side that appears at some resolutions.

Initially I thought it was an issue with Render Resolution option being set at 2x, but as you can see in the video, at some resolutions the issue only appears at 1x and renders fine at 2x, and sometimes only gets fixed at 1.5x!

I also wondered if maybe it was something to do with 2880x2160 and 3200x1800 being custom resolutions that I created in Nvidia Control Panel, but 2560x1600 and 2560x1440 are standard resolutions that exhibit the issue.

The outline appears to be more pronounced at higher resolutions.

RTX 2060 Super on driver 560.94

**Comment by Polymega** [comment]
I am likely going to butcher the explanation as to why, but I'll give it a shot. 😅 

In order for our soft shadows to retain the same "softness," regardless of resolution being used, we have to do some behind-the-scenes math with how the soft shadows are rendered. When we restored soft shadows, we used 720p as a base for its softness. Then, for any other resolution, we run a formula to keep that softness level. 

It'd roughly be something like: 720/ [resolution height] = A, A * [specific value] = B, B = [new value to maintain same softness]

Anyway, certain resolutions will divide extra-funny into 720, create an unusual final value for the game to use to determine soft shadow behavior. What you shared here is an example of that. When it happens, the shadows appear "slightly misaligned", for a lack of better term.

It's unfortunately something we can't fully rectify. We have gone back to tighten up the formula to make this less obvious or less likely to happen at certain resolutions, but it can still happen nonetheless.
